lib/practice_exercises.rb

```diff
@@ -1,13 +1,63 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: n log(n)
+# Space Complexity: n
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  #raise NotImplementedError, "Not implemented yet"
+
+  pattern = /[A-Za-z]/
+  all_chars = string.chars
+
+  concat = []
+
+  all_chars.each do |character|
+    if pattern.match(character)
+      concat << character.downcase
+    end
+  end
+
+  if concat.length == 0 or concat.length == 1
+    return true
+  end
+
+  i = 0
+  j = concat.length - 1
+
+  until j<=i do
+    if concat[i] != concat[j]
+      return false
+    end
+    i += 1
+    j -= 1
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O = nm
+# Space Complexity: O = nm
 def longest_prefix(strings)
```

Comment:
Since you are sorting the time complexity is O(n log n + nm).  Do you need to sort?